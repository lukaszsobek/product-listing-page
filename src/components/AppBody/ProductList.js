import React from "react";

const ProductList = props => {
    const { productList } = props;

    const listItems = productList.map((productData, key) => {
        return <ProductListItem key={key} data={productData} />
    });

    return (
        <section className="product-list">
            { listItems }
        </section>
    );
};

const ProductListItem = props => {
    const { name, category, colors, price } = props.data;

    return (
        <div className="product-list__item">
            <div className="product-list__item__name">{ name }</div>
            <div className="product-list__item__category">Category: { category }</div>
            <div className="product-list__item__colors">
                Colors: <ColorList data={colors} />
            </div>
            <div className="product-list__item__price">Price: &euro; { price }</div>
        </div>
    );
}

const ColorList = props => {
    return props.data.map((color, key) => {
        return (
            <span key={key} className="product-list__item__colors__name" >
                { color }
            </span>
        );
    });
};

export { ColorList, ProductListItem, ProductList };
export default ProductList;