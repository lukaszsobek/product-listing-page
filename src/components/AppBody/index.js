import React from "react";

import ProductList from "./ProductList";

const AppBody = props => {
    const { productList } = props;
    
    return (
        <main>
            <ProductList productList={productList}/>
        </main>
    );
}

export default AppBody;