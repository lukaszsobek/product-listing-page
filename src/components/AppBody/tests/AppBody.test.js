import AppBody from '../'

describe('AppBody', () => {

  // error
  it('renders with error', () => {
    const props = {
      loadedState: "error"
    }
    expect(AppBody(props)).toMatchSnapshot()
  })

  // loading
  it('renders when loading', () => {
    const props = {
      loadedState: "loading"
    }
    expect(AppBody(props)).toMatchSnapshot()
  })

  it('renders with props', () => {
    const props = {
      productList: [1, 2]
    }
    expect(AppBody(props)).toMatchSnapshot()
  })
})
