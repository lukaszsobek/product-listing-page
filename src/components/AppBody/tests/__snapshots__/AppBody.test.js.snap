// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody renders when loading 1`] = `
<div
  className="products--loading"
>
  Loading products...
</div>
`;

exports[`AppBody renders with error 1`] = `
<div
  className="products--error"
>
  Error loading products from remote server.
</div>
`;

exports[`AppBody renders with props 1`] = `
<main>
  <ProductList
    productList={
      Array [
        1,
        2,
      ]
    }
  />
</main>
`;
